sdk/python/src/dagger/telemetry/__init__.py

```diff
@@ -0,0 +1,185 @@
+from collections.abc import Callable
+import logging
+import os
+from typing import Final, Literal
+
+from opentelemetry import context, propagate
+from opentelemetry.environment_variables import OTEL_PYTHON_TRACER_PROVIDER
+from opentelemetry.sdk._configuration import (
+    _init_logging,
+    _init_metrics,
+    _import_exporters,
+    _get_exporter_names,
+)
+from opentelemetry.sdk.environment_variables import (
+    OTEL_EXPORTER_OTLP_ENDPOINT,
+    OTEL_EXPORTER_OTLP_INSECURE,
+    OTEL_EXPORTER_OTLP_METRICS_ENDPOINT,
+    OTEL_EXPORTER_OTLP_METRICS_INSECURE,
+    OTEL_EXPORTER_OTLP_LOGS_ENDPOINT,
+    OTEL_EXPORTER_OTLP_LOGS_INSECURE,
+    OTEL_EXPORTER_OTLP_TRACES_ENDPOINT,
+    OTEL_EXPORTER_OTLP_TRACES_INSECURE,
+    OTEL_SDK_DISABLED,
+    OTEL_SERVICE_NAME,
+)
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.sdk.trace.export import SpanExporter
+from opentelemetry.trace import get_tracer_provider, propagation
+
+from opentelemetry import trace
+from opentelemetry.sdk._configuration import _BaseConfigurator as _BaseSDKConfigurator
+from opentelemetry.semconv.trace import SpanAttributes
+
+__all__ = [
+    "initialize",
+    "get_tracer",
+    "otel_configured",
+    "otel_enabled",
+]
+
+SERVICE_NAME: Final = "dagger-python-sdk"
+
+logger = logging.getLogger(__name__)
+
+
+def initialize():
+    """Configure telemetry."""
+    _DaggerPropagationConfigurator().configure()
+    _DaggerOtelConfigurator().configure()
+
+
+def get_tracer() -> trace.Tracer:
+    """Returns a tracer to use with Dagger."""
+    initialize()
+    return trace.get_tracer(
+        "dagger.io/sdk.python",
+        schema_url=SpanAttributes.SCHEMA_URL,
```

Comment:
Any idea what semconv version we're on here? I'm guessing we'll need to keep it in sync across all SDKs, currently it's 1.24.0. Not completely sure. I've seen cryptic warnings when there's a mismatch, so kind of dreading the day it comes up again.